### ISSUE TITLE
Request signature is incorrect when using credentials with a token

### ISSUE THREAD

**Issue by thebogusman** [AllocZero/EfficientDynamoDb#117]
When I use temporary AWS credentials (AccessKey+SecretKey+Token), obtained for example by assuming a role in CLI:
`aws sts assume-role --role-arn arn:aws:iam::000000000000:role/my-custom-role --role-session-name my-session`

all requests fail with a bad signature error:
```
Response:
StatusCode: 400, ReasonPhrase: 'Bad Request', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Server: Server
  Date: Mon, 12 Apr 2021 20:55:19 GMT
  Connection: keep-alive
  x-amzn-RequestId: VVCRE16RCALHED6VBRGTNBUHCJVV4KQNSO5AEMVJF66Q9ASUAAJG
  x-amz-crc32: 1393778603
  Content-Type: application/x-amz-json-1.0
  Content-Length: 254
}
{"__type":"com.amazon.coral.service#InvalidSignatureException","message":"The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details."}
```

What I'm actually trying to do is to use the library inside a lambda function with lambda invocation role's credentials, however it fails due to the aforementioned issue. Here's how I'm obtaining the credentials (FallbackCredentialsFactory comes from AWSSDK.Core):

```
var credentials = await FallbackCredentialsFactory.GetCredentials().GetCredentialsAsync();
var efficientCredentials = new AwsCredentials(
	credentials.AccessKey,
	credentials.SecretKey,
	credentials.UseToken ? credentials.Token : null
);
```

In both cases AmazonDynamoDBClient works fine with the same credentials.

Any idea what might be wrong?

**Comment by lezzi** [comment]
@thebogusman are you currently using `0.9.0`? We have addressed this problem in `0.9.1` alpha

**Comment by thebogusman** [comment]
Yes and switching to `0.9.1` has fixed the issue :) Thank you!
